sycl/test-e2e/AOT/multiple-devices.cpp

```diff
@@ -6,7 +6,7 @@
 //
 //===----------------------------------------------------------------------===//
 
-// REQUIRES: opencl-aot, ocloc, cpu, gpu, accelerator
+// REQUIRES: opencl-aot, ocloc, any-device-is-cpu, any-device-is-gpu, any-device-is-accelerator
```

Comment:
I don't quite understand how it is a correct fix. If `,` is treated as `&&` in `REQUIRES` directive, then we previously had: `cpu && gpu && acc` which is never true. Now we have `any-device-is-cpu && any-device-is-gpu && any-device-is-accelerator`.

If the expression is computed per single device, then even with the latter there will only be `true` for just one of `any-device-is`, meaning the overall expression evaluates to `false` anyways.

Should we just drop those device requirements from the test and instead use `UNSUPPORTED` to filter out environments where the test can't be run (like CUDA)?

Considering that there are no device types other than CPU, GPU and ACC, do we ever need to require them all? My understanding is that it is safe to assume that `check-sycl-e2e` is always launched with an environment with at least one device. And if it is not the case yet, we should maybe introduce a check like this to `lit.cfg.py`

Reply:
Old features are per-device, new features are per-llvm-lit invocation. They were created to target this scenario exactly.